src/async_api/reply.rs

```diff
@@ -0,0 +1,1336 @@
+        Err(before_epoch_error) => (
+            -(before_epoch_error.duration().as_secs() as i64),
+            before_epoch_error.duration().subsec_nanos(),
+        ),
+    }
+}
+
+// Some platforms like Linux x86_64 have mode_t = u32, and lint warns of a trivial_numeric_casts.
+// But others like macOS x86_64 have mode_t = u16, requiring a typecast.  So, just silence lint.
+#[allow(trivial_numeric_casts)]
+/// Returns the mode for a given file kind and permission
+fn mode_from_kind_and_perm(kind: FileType, perm: u16) -> u32 {
+    (match kind {
+        FileType::NamedPipe => S_IFIFO,
+        FileType::CharDevice => S_IFCHR,
+        FileType::BlockDevice => S_IFBLK,
+        FileType::Directory => S_IFDIR,
+        FileType::RegularFile => S_IFREG,
+        FileType::Symlink => S_IFLNK,
+        FileType::Socket => S_IFSOCK,
+    }) as u32
+        | perm as u32
+}
+
+/// Returns a fuse_attr from FileAttr
+#[cfg(target_os = "macos")]
+fn fuse_attr_from_attr(attr: &FileAttr) -> fuse_attr {
+    let (atime_secs, atime_nanos) = time_from_system_time(&attr.atime);
+    let (mtime_secs, mtime_nanos) = time_from_system_time(&attr.mtime);
+    let (ctime_secs, ctime_nanos) = time_from_system_time(&attr.ctime);
+    let (crtime_secs, crtime_nanos) = time_from_system_time(&attr.crtime);
+
+    fuse_attr {
+        ino: attr.ino,
+        size: attr.size,
+        blocks: attr.blocks,
+        atime: atime_secs,
+        mtime: mtime_secs,
+        ctime: ctime_secs,
+        crtime: crtime_secs as u64,
+        atimensec: atime_nanos,
+        mtimensec: mtime_nanos,
+        ctimensec: ctime_nanos,
+        crtimensec: crtime_nanos,
+        mode: mode_from_kind_and_perm(attr.kind, attr.perm),
+        nlink: attr.nlink,
+        uid: attr.uid,
+        gid: attr.gid,
+        rdev: attr.rdev,
+        flags: attr.flags,
+        #[cfg(feature = "abi-7-9")]
+        blksize: attr.blksize,
+        #[cfg(feature = "abi-7-9")]
+        padding: attr.padding,
+    }
+}
+
+/// Returns a fuse_attr from FileAttr
+#[cfg(not(target_os = "macos"))]
+fn fuse_attr_from_attr(attr: &FileAttr) -> fuse_attr {
+    let (atime_secs, atime_nanos) = time_from_system_time(&attr.atime);
+    let (mtime_secs, mtime_nanos) = time_from_system_time(&attr.mtime);
+    let (ctime_secs, ctime_nanos) = time_from_system_time(&attr.ctime);
+
+    fuse_attr {
+        ino: attr.ino,
+        size: attr.size,
+        blocks: attr.blocks,
+        atime: atime_secs,
+        mtime: mtime_secs,
+        ctime: ctime_secs,
+        atimensec: atime_nanos,
+        mtimensec: mtime_nanos,
+        ctimensec: ctime_nanos,
+        mode: mode_from_kind_and_perm(attr.kind, attr.perm),
+        nlink: attr.nlink,
+        uid: attr.uid,
+        gid: attr.gid,
+        rdev: attr.rdev,
+        #[cfg(feature = "abi-7-9")]
+        blksize: attr.blksize,
+        #[cfg(feature = "abi-7-9")]
+        padding: attr.padding,
+    }
+}
```

Comment:
Instead of copying these, can you import them from the sync `reply.rs`?

Reply:
👍